bank_subscription_prediction/steps/data_loader.py

```diff
@@ -0,0 +1,56 @@
+import pandas as pd
+import os
+import requests
+from zenml import step
+
+@step
+def load_data(csv_file_path: str = "data/bank.csv") -> pd.DataFrame:
+    """Loads data from a CSV file or downloads it if not available.
+
+    Args:
+        csv_file_path: Path to the CSV file.
+
+    Returns:
+        Pandas DataFrame loaded from the CSV.
+    """
+    # Check if file exists locally
+    if not os.path.exists(csv_file_path):
+        print(f"File {csv_file_path} not found. Downloading from UCI ML Repository...")
+        
+        # URL for the bank marketing dataset
+        url = "https://archive.ics.uci.edu/ml/machine-learning-databases/00222/bank-additional.zip"
```

Comment:
maybe have this as a constant somewhere and/or log it as metadata?

Reply:
just it's useful for the user to have visible. parameter is fine though I guess user wouldn't be switching that out so often.